src/coreclr/tools/Common/TypeSystem/Common/MetadataVirtualMethodAlgorithm.cs

```diff
@@ -936,7 +916,7 @@ public static MethodDesc ResolveVariantInterfaceMethodToStaticVirtualMethodOnTyp
             TypeDesc interfaceType = interfaceMethod.OwningType;
 
             // Search for match on a per-level in the type hierarchy
-            for (MetadataType typeToCheck = currentType; typeToCheck != null; typeToCheck = typeToCheck.MetadataBaseType)
+            for (TypeDesc typeToCheck = currentType; typeToCheck != null; typeToCheck = typeToCheck.BaseType)
```

Comment:
Why is this change needed?

Reply:
See above - reverted all changes to MetadataVirtualMethodAlgorithm per your feedback.